source/common/upstream/health_checker_impl.cc

```diff
@@ -497,8 +502,6 @@ void TcpHealthCheckerImpl::TcpActiveHealthCheckSession::onData(Buffer::Instance&
       expect_close_ = true;
       client_->close(Network::ConnectionCloseType::NoFlush);
     }
-  } else {
-    host_->setActiveHealthFailureType(Host::ActiveHealthFailureType::UNHEALTHY);
```

Comment:
Seems like there should be some handling of this case? Or is this dealt with elsewhere?

Reply:
This didn't make any sense. See https://github.com/envoyproxy/envoy/pull/14772/files#diff-f9d4ff37ae025ffb1d0a2a86dd366f10a90746bc8452f6ebb28740735ff6f6f7R3516-R3518. I will add more comments here.